### PR TITLE
input_common: Add more libusb error checks for GC Adapter

### PR DESCRIPTION
**Pull request by ameerj**
Some yuz-ers have reported crashes relating to libusb. This PR adds more error checking on libusb calls, logging errors as encountered and halting executions earlier when errors are encountered.

### CODE REVIEW

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 1]
```diff
@@ -215,6 +219,10 @@ void Adapter::Setup() {
 
     // populate the list of devices, get the count
     const std::size_t device_count = libusb_get_device_list(libusb_ctx, &devices);
```
Comment (bunnei) [inline src/input_common/gcadapter/gc_adapter.cpp:221]: This needs to be signed

Comment (bunnei) [reply src/input_common/gcadapter/gc_adapter.cpp:221]: e.g. ssize_t

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 2]
```diff
@@ -279,7 +288,13 @@ bool Adapter::CheckDeviceAccess(libusb_device* device) {
 
 void Adapter::GetGCEndpoint(libusb_device* device) {
     libusb_config_descriptor* config = nullptr;
-    libusb_get_config_descriptor(device, 0, &config);
+    const int config_descriptor_error = libusb_get_config_descriptor(device, 0, &config);
+    if (config_descriptor_error) {
```
Comment (bunnei) [inline src/input_common/gcadapter/gc_adapter.cpp:292]: Should this be compared against LIBUSB_SUCCESS?

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 3]
```diff
@@ -215,6 +219,10 @@ void Adapter::Setup() {
 
     // populate the list of devices, get the count
     const std::size_t device_count = libusb_get_device_list(libusb_ctx, &devices);
+    if (device_count < 0) {
+        LOG_ERROR(Input, "libusb_get_device_list failed with error: {}", device_count);
+        return;
+    }
```
Comment (bunnei) [inline src/input_common/gcadapter/gc_adapter.cpp:225]: Should we also check that devices != nullptr?

File: src/input_common/gcadapter/gc_adapter.cpp [hunk 4]
```diff
@@ -215,6 +219,10 @@ void Adapter::Setup() {
```
Comment (bunnei) [inline src/input_common/gcadapter/gc_adapter.cpp:219]: Please initialize devices above to `{}` so the memory is zero'd

File: src/input_common/gcadapter/gc_adapter.cpp (hunk 2 above)
Comment (ameerj) [reply src/input_common/gcadapter/gc_adapter.cpp:292]: `LIBUSB_SUCCESS` is 0, just a shorthand but I addressed this for better clarity.

**Pull request merged**